### PR TITLE
provide option to add an tc-container to weave-net

### LINKED ISSUE #7879: Provide an option to enable weave-tc workaround
**1. Description**

The Current netfilter implementation of the Linux Kernel has issues wich can cause in several DNS-Problems.
A good article about the problem(s) can be found in the [weaveworks-blog](https://www.weave.works/blog/racy-conntrack-and-dns-lookup-timeouts).
With this option we provide the (by weavework) [suggested workaround](https://www.weave.works/blog/racy-conntrack-and-dns-lookup-timeouts)
with [tc](https://linux.die.net/man/8/tc).

Currently it would be needed to patch the weave-net-daemonset on existing clusters (wich would cause a network-interruption on every node) and keep track of not being overwritten by an `kops update` / `kops rolling-update`. This is very risky on production-clusters.

**2. Feel free to provide a design supporting your feature request.**

I suggest to have an (disabled by default) option which enable the tc-workaround. The creator of the weave-tc workaround suggests to [place the needed container inside network-pod](https://github.com/Quentin-M/weave-tc#how-to-run-it).

The option could look like this:

```yaml
spec:
  networking:
    weave:
      enableWeaveTc: true
```

### PR DESCRIPTION
**Pull request by jochen42**
fixed #7879

### CODE REVIEW

File: upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template [hunk 1]
```diff
@@ -223,6 +223,17 @@ spec:
           volumeMounts:
             - name: xtables-lock
               mountPath: /run/xtables.lock
+        {{- if .Networking.Weave.EnableWeaveTc }}
+        - name: weave-tc
+          image: 'qmachu/weave-tc:bd94b89'
```
[the remark this answers is not in the document]
Comment (jochen42) [reply upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template]: hi @rifelpet ,

you can find github-repo with the Dockerfile and the documentation here:
https://github.com/Quentin-M/weave-tc

If this is not enough, can you please provide me an link to the documentation about the security-rules?

thx, jochen!

[the remark this answers is not in the document]
Comment (jochen42) [reply upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template]: hi @recollir ,

and yes it is the suggestion of weaveworks. i already-descriped it in the issue #7879.
You can read it in the following article at weaveworks blog in the part "Suggestions":

https://www.weave.works/blog/racy-conntrack-and-dns-lookup-timeouts

thx, jochen

[the remark this answers is not in the document]
Comment (rifelpet) [reply upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template]: I agree that the solution is what weaveworks suggests, my only concern is the choice of docker image itself. Does weaveworks suggest using this docker image specifically? Is this image mentioned in their documentation? It appears rather "unofficial", in that it is more difficult to trust that the image isnt compromised, compared to if it was under the official [weaveworks dockerhub account](https://hub.docker.com/u/weaveworks). Alternatively we could build an equivalent image ourselves within Kops and push it to the same docker repos that we push our other images. I'm not sure it's worth the effort though.

Anyone else have an opinion on this?

[the remark this answers is not in the document]
Comment (jochen42) [reply upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template]: I aggree! Weaveworks offical account would be better and a auditable pipeline/dockerrepo inside kops would be the best solution.

I am totally new in contributing to kops, but if anybody give's me an hint how to integrate it here, i would invest this effort.

**Comment by jochen42** [comment]
won't work like this due different host path's on different distributions. i will close the pr.

**Pull request closed without merging**